### PR TITLE
[Java] New Java actor API

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

1) As discussed in #7291, replace Java actor API with this new style. See `ActorTest::testCreateAndCallActor` for example.
2) Remove the `@RayRemote` annotation.
3) Rename `call` to `callActor` in `RayRuntime`.

Docs will be updated in a different PR.

## Related issue number

#7291

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: java/runtime/src/main/java/org/ray/runtime/util/generator/RayCallGenerator.java [hunk 1]
```diff
@@ -8,18 +8,18 @@
...
+ * A util class that generates `RayCall.java` and `ActorCall.java`, which provide type-safe
+ * interfaces for `Ray.call`, `Ray.createActor` and `actor.call`.
  */
 public class RayCallGenerator extends BaseGenerator {
 
   /**
    * @return Whole file content of `RayCall.java`.
    */
-  private String build() {
+  private String generateRayCallDotJava() {
```
Comment (zhijunfu) [inline java/runtime/src/main/java/org/ray/runtime/util/generator/RayCallGenerator.java]: Maybe `generateRayCallJava`?

Comment (raulchen) [reply java/runtime/src/main/java/org/ray/runtime/util/generator/RayCallGenerator.java]: I thought about that. But that seemed more vague. So I ended up with this name. But, anyway, I think the documentation is clear enough.

File: java/api/src/main/java/org/ray/api/RayActor.java [hunk 2]
```diff
@@ -5,12 +5,17 @@
...
-public interface RayActor<T> {
+public interface RayActor<A> extends ActorCall<A> {
 
   /**
    * @return The id of this actor.
    */
   ActorId getId();
+
+  @Override
+  default RayActor<A> getThis() {
```
Comment (kfstorm) [inline java/api/src/main/java/org/ray/api/RayActor.java]: Can we avoid this? It should not be exposed as an API.

Comment (raulchen) [reply java/api/src/main/java/org/ray/api/RayActor.java]: yeah, it's a bit weird here. I'll move it to the impl class.

**Review by zhijunfu — APPROVED** [review]
LGTM

**Review by kfstorm — APPROVED** [review]
LGTM.

**Pull request merged**